Workflow failure: Keycloak CI - OAuth 2.0 Grant Type SPI

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

https://github.com/keycloak/keycloak/actions/runs/8058248344/attempts/1

Failed with next message:
```
CC [M]  /home/runner/work/keycloak/keycloak/.github/fake_fips/fake_fips.o
/home/runner/work/keycloak/keycloak/.github/fake_fips/fake_fips.c:42:18: error: ‘struct ctl_table’ has no member named ‘child’
   42 |                 .child          = crypto_sysctl_table
      |                  ^~~~~
/home/runner/work/keycloak/keycloak/.github/fake_fips/fake_fips.c:42:35: error: positional initialization of field in ‘struct’ declared with ‘designated_init’ attribute [-Werror=designated-init]
   42 |                 .child          = crypto_sysctl_table
      |                                   ^~~~~~~~~~~~~~~~~~~
/home/runner/work/keycloak/keycloak/.github/fake_fips/fake_fips.c:42:35: note: (near initialization for ‘crypto_dir_table[0]’)
/home/runner/work/keycloak/keycloak/.github/fake_fips/fake_fips.c:42:35: error: incompatible types when initializing type ‘enum <anonymous>’ using type ‘struct ctl_table *’
/home/runner/work/keycloak/keycloak/.github/fake_fips/fake_fips.c: In function ‘crypto_proc_fips_init’:
/home/runner/work/keycloak/keycloak/.github/fake_fips/fake_fips.c:51:26: error: implicit declaration of function ‘register_sysctl_table’; did you mean ‘unregister_sysctl_table’? [-Werror=implicit-function-declaration]
   51 |         crypto_sysctls = register_sysctl_table(crypto_dir_table);
      |                          ^~~~~~~~~~~~~~~~~~~~~
      |                          unregister_sysctl_table
/home/runner/work/keycloak/keycloak/.github/fake_fips/fake_fips.c:51:24: warning: assignment to ‘struct ctl_table_header *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
   51 |         crypto_sysctls = register_sysctl_table(crypto_dir_table);
      |                        ^
cc1: some warnings being treated as errors
make[3]: *** [scripts/Makefile.build:251: /home/runner/work/keycloak/keycloak/.github/fake_fips/fake_fips.o] Error 1
make[2]: *** [/usr/src/linux-headers-6.5.0-1015-azure/Makefile:[20](https://github.com/keycloak/keycloak/actions/runs/8058248344/job/22010977165#step:3:21)39: /home/runner/work/keycloak/keycloak/.github/fake_fips] Error 2
make[1]: *** [Makefile:[23](https://github.com/keycloak/keycloak/actions/runs/8058248344/job/22010977165#step:3:24)4: __sub-make] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-6.5.0-1015-azure'
make: *** [Makefile:4: all] Error 2
Error: Process completed with exit code 2.
``` 

### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

--- a/.github/fake_fips/fake_fips.c
+++ b/.github/fake_fips/fake_fips.c
@@ -22,6 +22,7 @@
 
 #include <linux/module.h>
 #include <linux/sysctl.h>
+#include <linux/version.h>
 
 int fips_enabled = 1;
 
@@ -39,7 +40,9 @@ static struct ctl_table crypto_dir_table[] = {
 	{
 		.procname       = "crypto",
 		.mode           = 0555,
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0))
 		.child          = crypto_sysctl_table
+#endif
 	},
 	{}
 };
@@ -48,7 +51,11 @@ static struct ctl_table_header *crypto_sysctls;
 
 static void crypto_proc_fips_init(void)
 {
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0))
 	crypto_sysctls = register_sysctl_table(crypto_dir_table);
+#else
+	crypto_sysctls = register_sysctl(crypto_dir_table->procname, crypto_sysctl_table);
+#endif
 }
 
 static void crypto_proc_fips_exit(void)

